corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -604,7 +603,6 @@ def set_archived_state(form, archive, user_id):
         form.state = XFormInstanceSQL.ARCHIVED if archive else XFormInstanceSQL.NORMAL
 
     @staticmethod
-    @transaction.atomic
```

Comment:
How come these are not needed to be in transaction anymore?

Reply:
This transaction was on the `default` database which is quite pointless since forms and cases aren't stored in that DB.

This does already use an atomic block in the function on the correct DB.